### PR TITLE
Issue 1009: Added redirect (303) handling to sync jobs.

### PR DESCRIPTION
**Pull request by jcsegovia**
Fix for issue 1009. Added redirect handling to sync. jobs.

### CODE REVIEW

File: astroquery/utils/tap/core.py [hunk 1]
```diff
@@ -207,6 +207,20 @@ def launch_job(self, query, name=None, output_file=None,
...
+        # handle redirection
+        if response.status == 303:
+            # redirection
+            if verbose:
+                print("Redirection found")
+            location = self.__connHandler.find_header(
+                response.getheaders(),
+                "location")
+            if location is None:
+                raise Exception("No location found after redirection was received (303)")
```
Comment (keflavich) [inline astroquery/utils/tap/core.py]: Could you make this a specific exception type?  Perhaps `requests.exceptions.HTTPError`

**Comment by keflavich** [comment]
pycodestyle is complaining about spaces in the comments.  Would you mind fixing those?  In the test code, the commends are formatted as `#blah` but the standard we have adopted is `# blah`

**Comment by jcsegovia** [comment]
@keflavich do you know why my local pep8 does not show any of the reported errors (E123/E265)...?

**Comment by keflavich** [comment]
@jcsegovia maybe it has to do with some local config file?  I don't know for sure though.

Looks like we're down to just the known & unrelated CI bug.  So I will merge.

**Pull request merged**